app/v2/coinswap/internal/keeper/swap.go

```diff
@@ -73,10 +73,10 @@ func (k Keeper) tradeExactInputForOutput(ctx sdk.Context, input types.Input, out
 	if err != nil {
 		return sdk.ZeroInt(), err
 	}
-	// assert that the calculated amount is less than the
+	// assert that the calculated amount is more than the
 	// minimum amount the buyer is willing to buy.
 	if boughtTokenAmt.LT(output.Coin.Amount) {
-		return sdk.ZeroInt(), types.ErrConstraintNotMet(fmt.Sprintf("token amount (%s) to be bought was less than the minimum amount (%s)", boughtTokenAmt, output.Coin.Amount))
+		return sdk.ZeroInt(), types.ErrConstraintNotMet(fmt.Sprintf("the actual amount (%s) of tokens (%s) to be bought was less than the minimum amount (%s) of buyer bought", boughtTokenAmt, output.Coin.Denom, output.Coin.Amount))
```

Comment:
the actual amount (%s) of tokens (%s) to be bought is less than the minimum amount (%s) the buyer is willing to buy